hedera-node/test-clients/src/test/java/com/hedera/services/bdd/utils/test/translators/UtilPrngTranslatorTest.java

```diff
@@ -0,0 +1,127 @@
+        when(mockTransactionBlockItems.output().hasUtilPrng()).thenReturn(true);
+        when(mockTransactionBlockItems.output().utilPrng()).thenReturn(mockUtilPrngOutput);
+        final Bytes bytes = Bytes.fromHex("badcadfaddad2bedfedbeef959feedbeadcafecadecedebeed4acedecada5ada");
+        when(mockUtilPrngOutput.entropy()).thenReturn(new OneOf<>(UtilPrngOutput.EntropyOneOfType.PRNG_BYTES, bytes));
+
+        SingleTransactionRecord result = translator.translate(mockTransactionBlockItems, mockStateChanges);
+
+        // Then
+        TransactionRecord expectedRecord =
+                TransactionRecord.newBuilder().prngBytes(bytes).build();
+
+        assertEquals(mockTransaction, result.transaction());
+        assertEquals(expectedRecord, result.transactionRecord());
+        assertEquals(result.transactionSidecarRecords(), List.of());
+        assertEquals(result.transactionOutputs(), new SingleTransactionRecord.TransactionOutputs(null));
+    }
+
+    @Test
+    void testTranslateWithPrngNumber() {
+        // When
+        when(mockTransactionBlockItems.txn()).thenReturn(mockTransaction);
+        when(mockTransactionBlockItems.output()).thenReturn(mockTransactionOutput);
+        when(mockTransactionBlockItems.output().hasUtilPrng()).thenReturn(true);
+        when(mockTransactionBlockItems.output().utilPrng()).thenReturn(mockUtilPrngOutput);
+        final int number = 42;
+        when(mockUtilPrngOutput.entropy()).thenReturn(new OneOf<>(UtilPrngOutput.EntropyOneOfType.PRNG_NUMBER, number));
+
+        SingleTransactionRecord result = translator.translate(mockTransactionBlockItems, mockStateChanges);
+
+        // Then
+        TransactionRecord expectedRecord =
+                TransactionRecord.newBuilder().prngNumber(number).build();
+
+        assertEquals(mockTransaction, result.transaction());
+        assertEquals(expectedRecord, result.transactionRecord());
+        assertEquals(result.transactionSidecarRecords(), List.of());
+        assertEquals(result.transactionOutputs(), new SingleTransactionRecord.TransactionOutputs(null));
+    }
+
+    @Test
+    void testTranslateNoPrng() {
+        // When
+        when(mockTransactionBlockItems.txn()).thenReturn(mockTransaction);
+        when(mockTransactionBlockItems.output()).thenReturn(mockTransactionOutput);
+        when(mockTransactionBlockItems.output().hasUtilPrng()).thenReturn(false);
+
+        SingleTransactionRecord result = translator.translate(mockTransactionBlockItems, mockStateChanges);
+
+        // Then
+        TransactionRecord expectedRecord = TransactionRecord.DEFAULT;
+
+        assertEquals(mockTransaction, result.transaction());
+        assertEquals(expectedRecord, result.transactionRecord());
+        assertEquals(result.transactionSidecarRecords(), List.of());
+        assertEquals(result.transactionOutputs(), new SingleTransactionRecord.TransactionOutputs(null));
+    }
```

Comment:
It may be good to also have unit tests for any non-obvious empty or default values that `UtilPrngOutput.EntropyOneOfType.PRNG_BYTES` and `UtilPrngOutput.EntropyOneOfType.PRNG_NUMBER` could have, e.g. `Bytes.EMPTY`

Reply:
Added, thanks!